infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -385,12 +356,16 @@ public void startSnapshotSyncApply() {
     }
 
     /**
-     * Clear local updated streams for which no data was transferred from active
-     * (hence were not cleared on the replication/transfer path)
+     * Clear streams which :
+     * 1.has evidenced data from active
+     * 2.has had local updates when the role was not STANDBY
      *
      * Note: streams could be locally written while this node had no assigned role.
      */
     public void clearLocalStreams() {
+        Set<UUID> streamsToClear = new HashSet<>();
+        streamsToClear.addAll(replicatedStreamIds);
```

Comment:
clearing the replicated streams here will still not make the (clear + apply) atomic.  IMO, we should clear the replicated streams in applyShadowStream() within the same tx as updateLog.

https://github.com/CorfuDB/CorfuDB/blob/master/infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java#L348

Reply:
True, but this change is only to bring the clearing of streams to apply phase.

Also, by clearing the stream in applyShadowStream(), we make only clearing+apply of 1 stream atomic at a time. It doesn't help with the use case where we don't want the client to observe the inconsistent state right..? (the DB will still be inconsistent when snapshot is in progress. We need to come up with a way to make clear+apply of all streams atomic, right?)

Reply:
My understanding was for `replicatedStreams`, we can do the "clear+apply" atomically for each stream in APPLY phase. After that, we iterate the `streamsWithLocalWrites` set and clear them all at once. Is that correct?

Reply:
Discussed offline as Pankti mentioned.